actionmailer/lib/action_mailer/test_helper.rb

```diff
@@ -1,13 +1,16 @@
 # frozen_string_literal: true
 
 require "active_job"
+require_relative "test_adapter"
 
 module ActionMailer
   # Provides helper methods for testing Action Mailer, including #assert_emails
   # and #assert_no_emails.
   module TestHelper
     include ActiveJob::TestHelper
 
+    DELIVERY_JOB = Object.new
```

Comment:
I thought about the lambda filter and had doubts so I went for an implementation that don't reach into the ActiveJob boundaries. My doubt was about the usefulness of supporting lambdas by applications.
If we decide to add this option to help fix the ActionMailer issue this will become public API, and again I wasn't sure if that's something developers will benefit from. Thoughts?